jwst/resample/resample_spec.py

```diff
@@ -313,6 +315,91 @@ def _max_virtual_slit_extent(self, wcs_list, target_ra, target_dec):
 
         return y_slit_min, y_slit_max
 
+    def build_miri_lrs_output(self):
+        """ Build MIRI LRS FIXEDSLIT output WCS. """
+        all_wavelength = []
+        all_ra_slit = []
+        all_dec_slit = []
+
+        for im, model in enumerate(self.input_models):
+            wcs = model.meta.wcs
+            bbox = wcs.bounding_box
+            grid = wcstools.grid_from_bounding_box(bbox)
+            ra, dec, lam = np.array(wcs(*grid))
+            # Handle vertical (MIRI) or horizontal (NIRSpec) dispersion.  The
+            # following 2 variables are 0 or 1, i.e. zero-indexed in x,y WCS order
+            spectral_axis = find_dispersion_axis(model)
+            spatial_axis = spectral_axis ^ 1
+
+            # Compute the wavelength array, trimming NaNs from the ends
+            # In many cases, a whole slice is NaNs, so ignore those warnings
+            warnings.simplefilter("ignore")
+            wavelength_array = np.nanmedian(lam, axis=spectral_axis)
+            warnings.resetwarnings()
+            wavelength_array = wavelength_array[~np.isnan(wavelength_array)]
+
+            all_wavelength = np.append(all_wavelength, wavelength_array)
+            this_minw = np.min(wavelength_array)
+            this_maxw = np.max(wavelength_array)
+            all_minw = np.min(all_wavelength)
+            all_maxw = np.max(all_wavelength)
+            if this_minw < all_minw:
+                addpts = wavelength_array[wavelength_array < all_minw]
+                all_wavelength = np.append(all_wavelength, addpts)
+            if this_maxw > all_maxw:
+                addpts = wavelength_array[wavelength_array > all_maxw]
+                all_wavelength = np.append(all_wavelength, addpts)
+
+        # done looping over set of models
+        all_wave = np.hstack(all_wavelength)
+        all_wave = all_wave[~np.isnan(all_wave)]
+        all_wave = np.sort(all_wave, axis=None)
+        # Tabular interpolation model, pixels -> lambda
+        wavelength_array = np.unique(all_wave)
+        # Check if the data is MIRI LRS FIXED Slit. If it is then
+        # the wavelength array needs to be flipped so that the resampled
+        # dispersion direction matches the dispersion direction on the detector.
```

Comment:
Isn't the "upside-down" dispersion direction for LRS FS already encoded by setting dispaxis=-2 (i.e. *negative* 2) in the meta data? Doesn't that somehow automatically get accounted for when constructing the spectral WCS axis and associated wavelength array?

Reply:
This is code related to the wavelength axis computation which I kept from the previous implementation. For this data 
```
cal.meta.wcsinfo.dispersion_direction
Out[43]: 2
```

Reply:
Huh, you're right. In all the post-assign_wcs products for LRS FS it says "DISPAXIS=2". I could've sworn we used signed integers for that keyword to document the *direction* of increasing wavelength, along with the axis. But I guess not.